icu4c/source/i18n/rbnf.cpp

```diff
@@ -1166,8 +1175,17 @@ RuleBasedNumberFormat::format(const DecimalQuantity &number,
 
             // TODO this section should probably be optimized. The DecimalFormat is shared in ICU4J.
             NumberFormat *decimalFormat = NumberFormat::createInstance(locale, UNUM_DECIMAL, status);
+            if (decimalFormat == nullptr) {
```

Comment:
Again,` LocalPointer<NumberFormat>` would be simpler.

Reply:
Thanks Andy!  This should be resolved in the latest commit.